### PR TITLE
Non Transactinal connection Injection

### PR DESCRIPTION
**Pull request by tushar-oracle**
Application Changes for the Non-Transaction Connection Changes

### CODE REVIEW

File: xa/java/department-spring-jpa/pom.xml [hunk 1]
```diff
@@ -67,27 +67,32 @@
...
 		<dependency>
 			<groupId>com.oracle.database.security</groupId>
 			<artifactId>osdt_cert</artifactId>
-			<version>21.3.0.0</version>
+			<version>21.9.0.0</version>
+		</dependency>
+		<dependency>
+			<groupId>javax.xml.bind</groupId>
+			<artifactId>jaxb-api</artifactId>
+			<version>2.3.0</version>
```
Comment (pruthvithej) [inline xa/java/department-spring-jpa/pom.xml:95]: Why do we need this dependency?  In spring boot we are using the Jakarta extension, if you need this then you most probably may need to add from groupId "jakarta.xml.bind"

File: xa/java/department-helidon-jpa/src/main/java/com/oracle/mtm/sample/data/AccountsService.java [hunk 2]
```diff
@@ -45,22 +45,14 @@ public class AccountsService implements IAccountsService {
     @TrmEntityManager
     private Provider<EntityManager> trmEntityManager;
 
-    @Inject
```
Comment (BharathMC) [inline xa/java/department-helidon-jpa/src/main/java/com/oracle/mtm/sample/data/AccountsService.java:48]: Have we removed this localEntityManager in Configuration class ?

- In all services

File: xa/java/department-spring-jpa/src/main/java/com/oracle/mtm/sample/XADataSourceConfig.java [hunk 3]
```diff
@@ -175,7 +175,7 @@ public EntityManagerFactory createEntityManagerFactory() throws SQLException {
         Properties properties = new Properties();
         properties.setProperty( "jakarta.persistence.transactionType", "RESOURCE_LOCAL"); // change this to resource_local
         properties.put("hibernate.show_sql", "true");
-        properties.put("hibernate.dialect", "org.hibernate.dialect.Oracle12cDialect");
```
Comment (BharathMC) [inline xa/java/department-spring-jpa/src/main/java/com/oracle/mtm/sample/XADataSourceConfig.java:178]: We don't need createEntityManagerFactory() method anymore right? if not used, let's remove

File: xa/java/department-helidon-jpa/src/main/java/com/oracle/mtm/sample/data/AccountsService.java (hunk 2 above)
Comment (tushar-oracle) [reply xa/java/department-helidon-jpa/src/main/java/com/oracle/mtm/sample/data/AccountsService.java:48]: Hi not all , as of now we decided to change only 4 application

File: xa/java/department-spring-jpa/pom.xml (hunk 1 above)
Comment (tushar-oracle) [reply xa/java/department-spring-jpa/pom.xml:95]: There was some Javax dependency issue in hibernate 6.3+. after resolving that I have removed it

**Review by pruthvithej — APPROVED** [review]

File: xa/java/department-spring-jpa/src/main/java/com/oracle/mtm/sample/XADataSourceConfig.java [hunk 4]
```diff
@@ -113,38 +110,15 @@ public DataSource getXADataSource() {
         return pds;
     }
 
-    public DataSource getDataSource() {
```
Comment (BharathMC) [inline xa/java/department-spring-jpa/src/main/java/com/oracle/mtm/sample/XADataSourceConfig.java:116]: We need to clean-up similar to this in `department-spring-jpa-multiplerm` and `department-spring-jpa-multiplerm-postgres` as well

Comment (tushar-oracle) [reply xa/java/department-spring-jpa/src/main/java/com/oracle/mtm/sample/XADataSourceConfig.java:116]: Bharath , we have discussed to only introduce these changes into 4 application of different types . 
Later we will introduce them to other application

**Pull request merged**